activerecord/lib/active_record/scoping/named.rb

```diff
@@ -183,6 +204,23 @@ def scope(name, body, &block)
               scope
             end
           end
+
+          if predicate
+            method_name = if predicate.is_a?(Symbol) || predicate.is_a?(String)
+              predicate.to_sym
+            elsif predicate == true
+              :"within_#{name}"
+            else
+              raise ArgumentError,
+                "Invalid value for 'predicate'. Expected a Symbol, " \
+                "a String, or true, but received #{predicate.inspect}"
+            end
+
+            define_method("#{method_name}?") do
+              self.class.send(name).exists?(id)
+            end
+          end
```

Comment:
I wonder if we could trim this a bit?

```suggestion
          predicate = :"within_#{name}" if predicate == true

          case predicate
          when nil # Skip it
          when Symbol, String
            define_method("#{predicate}?") do
              self.class.send(name).exists?(id)
            end
          else
            raise ArgumentError, "'predicate:' expected Symbol, String, or true, but got #{predicate.inspect}"
          end
```

Reply:
The original implementation looked like this, but raised a linting violation. 

https://github.com/rails/rails/blob/eb6615f14fdffe005c1d3be3c60341596fe7d4e0/activerecord/lib/active_record/scoping/named.rb#L208-L223

So then I tried this, but that also raised another linting violation:

https://github.com/rails/rails/blob/41308ee369632b545e5f5e6810966a09c75863e6/activerecord/lib/active_record/scoping/named.rb#L208-L228

So I just ended up with what you see now to suppress it.